### PR TITLE
Featuure/create recipe flow4

### PR DESCRIPTION
**Pull request by alexxcheung**

### CODE REVIEW

File: LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift [hunk 1]
```diff
@@ -625,6 +636,23 @@ class RecipeCreationViewController: UIViewController  {
...
     }
     
+    private func allTextFieldsAreEmpty() -> Bool {
+        if recipeNameTextField.text == "" && commentsTextView.text == "" && ingredientTextField.text == "" && amountTextField.text == "" && stepTextField.text == "" {
+            return true
+        }
+        return false
+    }
+    
+    private func updateIsRecipeEdited() {
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift]: Unreadable method name

File: LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift [hunk 2]
```diff
@@ -724,10 +752,27 @@ class RecipeCreationViewController: UIViewController  {
     }
     
     // MARK: Did Tap Buttons
-    @IBAction func didTapDone(_ sender: Any) {
+    @IBAction func didTapDoneButton(_ sender: Any) {
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift:755]: Why adding "Button"?

File: LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift [hunk 3]
```diff
@@ -724,10 +752,27 @@ class RecipeCreationViewController: UIViewController  {
...
         if editingMode {
-            self.presentDoneActionSheet()
+            
+            if editExistingRecipe {
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift]: That's a lot of if else and difficult to read, no possibility to refactor?

File: LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift [hunk 4]
```diff
@@ -973,8 +1018,9 @@ extension RecipeCreationViewController: UIImagePickerControllerDelegate, UINavig
     }
 }
 
-// MARK: TableView Configure
+// MARK: TableView Delegation
```
Comment (Ericordo) [inline LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift]: Delegate not delegation

**Review by Ericordo — APPROVED** [review]

File: LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift (hunk 2 above)
Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift:755]: Easier to search

File: LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift (hunk 3 above)
Comment (alexxcheung) [reply LetsDinnerV2 MessagesExtension/Controller/RecipeCreationVC/RecipeCreationViewController.swift]: updated

**Pull request merged**